### DISCUSSION TITLE
ArcFace - Age and gender detection values

### DISCUSSION THREAD

**Discussion by mnts-i** [exadel-inc/CompreFace#901] [General] [answered]
I am using InsightFace's custom build (and thus its custom age and gender detectors) and I noticed that the age detection returns the same values for high and low (instead of a range) and the gender detection often gives the wrong gender (with probability: 1) whereas FaceNet seems to be more precise. 

Does anyone else has the same results?

**Comment by pospielov** [comment, marked as answer]
InsightFace receives only one value for age detection. To make it compatible with FaceNet, we write the same value.
InsightFace and FaceNet use different approaches for age and gender detection. Both are not very accurate, especially compared with facial recognition.
We will integrate with more accurate libraries when they appear.
